managedcache/gvk.go

```diff
@@ -6,13 +6,23 @@ import (
 
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 )
 
 // ErrEmptyKindOrVersion is returned when either kind or version of the given object are empty.
 var ErrEmptyKindOrVersion = errors.New("object must have non-empty kind and version")
 
-func gvkForObject(obj runtime.Object) (schema.GroupVersionKind, error) {
+func gvkForObject(scheme *runtime.Scheme, obj runtime.Object) (schema.GroupVersionKind, error) {
```

Comment:
I was going to say "controller-runtime already has this function", and then saw that you're using it.

Why not have callers use `apiutil.GVKForObject` directly?

Reply:
fixed